packages/launch-editor/guess.js

```diff
@@ -24,16 +24,32 @@ module.exports = function guessEditor (specifiedEditor) {
   try {
     if (process.platform === 'darwin') {
       const output = childProcess
-        .execSync('ps x', {
+        .execSync('ps x -o comm=', {
           stdio: ['pipe', 'pipe', 'ignore']
         })
         .toString()
       const processNames = Object.keys(COMMON_EDITORS_OSX)
+      const processList = output.split('\n')
       for (let i = 0; i < processNames.length; i++) {
         const processName = processNames[i]
+        // Find editor by exact match.
         if (output.indexOf(processName) !== -1) {
           return [COMMON_EDITORS_OSX[processName]]
         }
+        const processNameWithoutApplications = processName.replace('/Applications', '')
+        // Find editor installation not in /Applications.
+        if (output.indexOf(processNameWithoutApplications) !== -1) {
+          // Use the CLI command if one is specified
+          if (processName !== COMMON_EDITORS_OSX[processName]) {
```

Comment:
If a partial match is found, but there is a CLI command for a full match then use the CLI.
For example, if VSCode is installed at `/Users/nrayburn/Applications/vscode.app`,  then it is likely the user still has `code` as a command on the CLI which should be used.

Reply:
Hey! I am using [IntelliJ IDEA Ultimate](https://www.jetbrains.com/fr-fr/idea/), installed via the [JetBrains ToolBox app](https://www.jetbrains.com/fr-fr/lp/toolbox/).

The executable name in my case is `/Users/grego/Applications/IntelliJ IDEA Ultimate.app/Contents/MacOS/idea`, so when using the global process name it does not work. I would advocate to use the running process instead.

<img width="952" alt="image" src="https://github.com/yyx990803/launch-editor/assets/51173502/50f4823d-a7c2-468c-89e9-06b009b1933a">
 
PR https://github.com/yyx990803/launch-editor/pull/66 does exactly that: it uses processNames from the `COMMON_EDITORS_OSX` list only if it's an exact match with the running process.

Moreover, https://github.com/yyx990803/launch-editor/pull/65 is needed for IntelliJ Ultimate users.

Thanks!